### PR TITLE
#157 Added first version of facet plus autodetection for XML files

### PR DESCRIPTION
**Pull request by adrianbumbas**

### CODE REVIEW

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @NotNull
+    @Override
+    public FacetType<CamelFacet, CamelFacetConfiguration> getFacetType() {
+        return CamelFacetType.getInstance();
+    }
+
+    @NotNull
+    @Override
+    public FileType getFileType() {
+        return StdFileTypes.XML;
```
Comment (davsclaus) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java]: I wonder if we can have ANY or something as file tile?

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @NotNull
+    @Override
+    public ElementPattern<FileContent> createSuitableFilePattern() {
+        return FileContentPattern.fileContent().with(new PatternCondition<FileContent>("") {
+            @Override
+            public boolean accepts(@NotNull final FileContent fileContent, final ProcessingContext context) {
```
Comment (davsclaus) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java]: And here we can check if the file is xml, java, kotlin, groovy or scala which is the files we support. And then also check if there is Camel inside the file, as you do.

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.jetbrains.annotations.NotNull;
+
+public class CamelFacetConfiguration implements FacetConfiguration {
+    @Override
+    public FacetEditorTab[] createEditorTabs(FacetEditorContext facetEditorContext, FacetValidatorsManager facetValidatorsManager) {
+        return new FacetEditorTab[]{new FacetEditorTab() {
+            @NotNull
+            @Override
+            public JComponent createComponent() {
+                return new JLabel("Content goes here");
```
Comment (davsclaus) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java]: I guess here is the UI for the facet where we can list the Camel routes discovered. And if the user clicks a route, it opens the file and position the cursor where the route starts.

Comment (adrianbumbas) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java]: Yes, this is the place for the facet UI. A facet is not meant to be used to navigate, it is for configuration only (see https://www.jetbrains.com/help/idea/2016.2/facet.html).
For navigation i think the best is to use a tool window like Spring provides.

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java (hunk 1 above)
Comment (adrianbumbas) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java]: There is nothing similar to ANY.

Comment (adrianbumbas) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFrameworkDetector.java]: I've ask the question on the Intellij community forum https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000031990-Facet-detector-for-multiple-file-types

File: camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java (hunk 3 above)
Comment (adrianbumbas) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java]: @davsclaus What would be the purpose of the facet? Configuration, navigation to the routes or both? If is for navigation only maybe we should look to other components and leave the facet aside.

Comment (davsclaus) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java]: @adrianbumbas yeah lets research a bit what IDEA offers and what is best. I wanted to have a window etc that scans your project and detect all your Camel routes, and list them, so you can see them in one window, and then click on the route to open the file in the source editor and position the cursor on the line where the route starts.

I guess facet is not the answer. If facet is better for project configuration then for Camel that makes less sense. If you use Camel with Spring Boot, the the SB facet is for configuring SB and Camel can do that too with SB configuration.

Comment (adrianbumbas) [inline camel-idea-plugin/src/main/java/org/apache/camel/idea/facet/CamelFacetConfiguration.java]: @davsclaus ok, let's put the facet aside. I think that the facet should exists at least for autodetection even there is no configuration, it gives a good feedback to the user. I will close this PR for now.

**Pull request closed without merging**